homeassistant/components/fully_kiosk/sensor.py

```diff
@@ -0,0 +1,115 @@
+    SensorEntityDescription(
+        key="batteryLevel",
+        name="Battery Level",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+    SensorEntityDescription(key="screenOrientation", name="Screen Orientation"),
+    SensorEntityDescription(
+        key="foregroundApp",
+        name="Foreground App",
+    ),
+    SensorEntityDescription(key="currentPage", name="Current Page"),
+    SensorEntityDescription(
+        key="wifiSignalLevel",
+        name="WiFi Signal Level",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+    ),
+    SensorEntityDescription(
+        key="internalStorageFreeSpace",
+        name="Internal Storage Free Space",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+    ),
+    SensorEntityDescription(
+        key="internalStorageTotalSpace",
+        name="Internal Storage Total Space",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+    ),
+    SensorEntityDescription(
+        key="ramFreeMemory",
+        name="RAM Free Memory",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+    ),
+    SensorEntityDescription(
+        key="ramTotalMemory",
+        name="RAM Total Memory",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+    ),
+)
+
+STORAGE_SENSORS = [
+    "internalStorageFreeSpace",
+    "internalStorageTotalSpace",
+    "ramFreeMemory",
+    "ramTotalMemory",
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Fully Kiosk Browser sensor."""
+    coordinator: FullyKioskDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    async_add_entities(
+        FullySensor(coordinator, description)
+        for description in SENSORS
+        if description.key in coordinator.data
+    )
+
+
+class FullySensor(FullyKioskEntity, SensorEntity):
+    """Representation of a Fully Kiosk Browser sensor."""
+
+    def __init__(
+        self,
+        coordinator: FullyKioskDataUpdateCoordinator,
+        sensor: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor entity."""
+        self.entity_description = sensor
+        self.coordinator = coordinator
+
+        self._attr_unique_id = f"{coordinator.data['deviceID']}-{sensor.key}"
+
+        super().__init__(coordinator)
+
+    @property
+    def native_value(self) -> Any:
+        """Return the state of the sensor."""
+        if (value := self.coordinator.data.get(self.entity_description.key)) is None:
+            return None
+
+        if self.entity_description.key in STORAGE_SENSORS:
+            return round(value * 0.000001, 1)
```

Comment:
Home Assistants should not calculate values (rounding we sometimes do, the others we like to avoid).